UnchainedLauncherGUI/src/App.xaml.cs

```diff
@@ -22,5 +30,53 @@ public App() : base() {
                 }
             }
         }
+        protected override void OnStartup(StartupEventArgs e) {
+            base.OnStartup(e);
+
+
+            // Init common dependencies
+            Octokit.GitHubClient githubClient = new Octokit.GitHubClient(new Octokit.ProductHeaderValue("UnchainedLauncher"));
+
+            var installationFinder = new Chivalry2InstallationFinder();
+            IUnchainedLauncherInstaller installer = new UnchainedLauncherInstaller(githubClient, Current.Shutdown);
+
+            // figure out if we need to install
+            var currentDirectory = Directory.GetParent(Assembly.GetExecutingAssembly().Location);
+            var needsInstallation = currentDirectory != null && !installationFinder.IsValidInstallation(currentDirectory);
+
+            // initialize the window
+            Window window = 
+                needsInstallation 
+                    ? InitializeInstallerWindow(installationFinder, installer) 
+                    : InitializeMainWindow(installationFinder, installer);
```

Comment:
Our case is kind of weird because we bundle the launcher and installer.  Because of this, we conditionally choose which one to initialize and show